src/array/cpu/spmm_blocking_libxsmm.h

```diff
@@ -0,0 +1,577 @@
+/*!
+ *  Copyright (c) 2021 Intel Corporation
+ * \file array/cpu/spmm.h
+ * \brief SPMM CPU kernel function header.
+ * \author Sanchit Misra <sanchit.misra@intel.com>,
+ *         Ramanarayan Mohanty <ramanarayan.mohanty@intel.com>,
+ *         Vasimuddin Md <vasimuddin.md@intel.com>,
+ *         Sasikanth Avancha <sasikanth.avancha@intel.com>
+ */
+#ifndef DGL_ARRAY_CPU_SPMM_BLOCKING_LIBXSMM_H_
+#define DGL_ARRAY_CPU_SPMM_BLOCKING_LIBXSMM_H_
+
+#include <dgl/array.h>
+#include <dgl/bcast.h>
+#include <dmlc/logging.h>
+#include <algorithm>
+
+#if !defined(_WIN32)
+#ifdef USE_AVX
+#ifdef USE_LIBXSMM
+#include <unistd.h>
+#include <libxsmm.h>
+#ifdef DEBUG
+#include <x86intrin.h>
+#endif
+#include <omp.h>
+
+namespace dgl {
+namespace aten {
+namespace cpu {
+
+template <typename IdType, typename DType>
+  struct CSRM{
+    IdType num_rows;
+    IdType num_cols;
+    IdType *indptr;
+    IdType *indices;
+    DType *data;
+  };
+
+int32_t get_llc_size() {
+  int32_t cache_size = sysconf(_SC_LEVEL3_CACHE_SIZE);
+  if(cache_size < 0) cache_size = DGL_CPU_LLC_SIZE;
+  return cache_size;
+}
+
+
+template <typename IdType, typename DType>
+inline void SpMMCreateBlocks(
+    const CSRMatrix& csr,
+    CSRM<IdType, IdType> *block_csr_array,
+    IdType num_M_blocks,
+    IdType num_K_blocks,
+    IdType M_BLOCK_SIZE,
```

Comment:
Variable names generally follow snake style.
```suggestion
    IdType m_block_size,
```

Reply:
Modified to M_block_size to emphasize that it is the block size corresponding to M.